external/fv3fit/fv3fit/reservoir/autoencoder.py

```diff
@@ -26,20 +41,195 @@ def encode(self, x: Union[np.ndarray, tf.Tensor]) -> np.ndarray:
     def decode(self, latent_x: Union[np.ndarray, tf.Tensor]) -> np.ndarray:
         return self.decoder.predict(latent_x)
 
+
+@io.register("dense-autoencoder")
+class AutoencoderPredictor(Reloadable):
```

Comment:
Suggestion: I suspect you can get the same functionality without adding this new class, or at least not as hefty a version of this class.

Instead of dumping and loading the encoder and decoder, can you just dump and load the `Autoencoder`, which is itself a `tf.keras.Model`? You could also remove the `input_variables` and `output_variables` attributes, which aren't used, though I'm curious how you'll make sure the stacking/packing/unpacking behavior is consistent between this and any later reservoir code which uses it.

If you keep input and output variables, this is just a `PureKerasModel`, where you can use `predictor.model.predict()` if you want to operate on ndarrays directly and you can ignore the `predictor.predict` method. If you are removing the input and output variables attributes, that's enough of a change that it's probably worth defining a `KerasModelOnly(Reloadable)` as a more general object, which simply dumps and loads a wrapped `self.model: tf.keras.Model` and has a `predict` as here.